builtin/credential/aws-ec2/path_login.go

```diff
@@ -429,6 +429,34 @@ func (b *backend) pathLoginUpdate(
 		return logical.ErrorResponse(fmt.Sprintf("Account ID %q does not belong to role %q", identityDocParsed.AccountID, roleName)), nil
 	}
 
+	// Verify that the `Region` of the instance trying to login matches the
+	// `Region` specified as a constraint on the role
+	if roleEntry.BoundRegion != "" && identityDocParsed.Region != roleEntry.BoundRegion {
+		return logical.ErrorResponse(fmt.Sprintf("Region %q does not belong to the role %q", identityDocParsed.Region, roleName)), nil
+	}
+
+	// Validate the SubnetID if corresponding bound was set on the role
+	if roleEntry.BoundSubnetID != "" {
+		subnetIDPtr := instanceDesc.Reservations[0].Instances[0].SubnetId
+		if subnetIDPtr == nil {
+			return nil, fmt.Errorf("Subnet ID in the instance description is nil")
+		}
+		if roleEntry.BoundSubnetID != *subnetIDPtr {
+			return logical.ErrorResponse(fmt.Sprintf("Subnet ID %q does not satisfy the constraint on the role %q", *subnetIDPtr, roleName)), nil
+		}
+	}
+
+	// Validate the VpcID if corresponding bound was set on the role
+	if roleEntry.BoundVpcID != "" {
```

Comment:
I realize Vault, and Go itself, are kind of inconsistent here but can this be BoundVPCID? I generally try to avoid camelcasing acronyms, even though the AWS variable does it.